blueman/gui/manager/ManagerDeviceList.py

```diff
@@ -315,6 +315,13 @@ def make_caption(name: str, klass: str, address: str) -> str:
         return "<span size='x-large'>%(0)s</span>\n<span size='small'>%(1)s</span>\n<i>%(2)s</i>" \
                % {"0": html.escape(name), "1": klass, "2": address}
 
+    @staticmethod
+    def make_display_name(alias: str, klass: int, address: str) -> str:
+        if alias.replace("-", ":") == address:
+            return _("Unnamed device")
```

Comment:
That means the same name for all unnamed devices, causing issues at least with "hide unnamed devices" disabled, but I guess there's more in your head as you're currently not using `klass`. :smile:

Reply:
I also experimented with naming it based on the device class description, but then I thought why as it's already shown. It can be any name, how about
 ```python
f"{alias} (Unnamed device)"
```

> causing issues at least with "hide unnamed devices" disabled

What issues specifically are you thinking of?

Reply:
Never mind, I forgot that we're showing the actual address as well, so it's not a mess of unnamed devices but you still see the addresses in any case.